src/Servers/Kestrel/test/InMemory.FunctionalTests/ChunkedResponseTests.cs

```diff
@@ -581,6 +628,65 @@ public async Task ChunksWithGetMemoryWithInitialFlushWorks()
             }
         }
 
+        [Fact]
+        public async Task ChunksWithGetMemoryBeforeFlushEdgeCase()
+        {
+            var length = new IntAsRef();
+            var semaphore = new SemaphoreSlim(initialCount: 0);
+            var testContext = new TestServiceContext(LoggerFactory);
+
+            using (var server = new TestServer(async httpContext =>
+            {
+                var response = httpContext.Response;
+
+                await response.StartAsync();
+
+                var memory = response.BodyWriter.GetMemory();
+                length.Value = memory.Length - 1;
```

Comment:
FYI: if length was a normal int, this test would still work since the closure would box it.